### DISCUSSION TITLE
How to compile this server written in go language into android running on armv7?

### DISCUSSION THREAD

**Discussion by furkanturkmn** [bluenviron/mediamtx#1473] [Questions]
**Hello,
In order to run this server by activity in android environment, I need to compile it for android running on armv7. Does anyone have any idea how I can do it?
_Thank you._**

**Comment by aler9** [comment]
Hello, in order to cross-compile the server for an armv7 device, from a machine with a different architecture (for instance amd64) it's enough to use the standard Golang command to cross-compile:

```
GOOS=linux GOARCH=arm GOARM=7 go build .
```

**Reply by furkanturkmn** [reply]
Thank you for your answer. I want to run this server in my own application. When I run it with JNI, the server is constantly closing and opening. How can I run this server in my own android software?

**Reply by UmutSDeniz** [reply]
> Hello, in order to cross-compile the server for an armv7 device, from a machine with a different architecture (for instance amd64) it's enough to use the standard Golang command to cross-compile:
> 
> ```
> GOOS=linux GOARCH=arm GOARM=7 go build .
> ```

Hello,

As your reply to OP I was trying to integrate the server into an android app by generating a library and starting the server through JNI call which is exposed over cgo. Inorder to build the server as a library I was using; `CGO_ENABLED=1 GOOS=android GOARCH=arm64 CC=/home/dev/Android-SDK/ndk/25.1.8937393/toolchains/llvm/prebuilt/linux-x86_64/bin/aarch64-linux-android33-clang go build -buildmode=c-shared -o libfoo.so main.go` as a result the server gets called but somehow it crashes and the android subsystem repeatedly tries to reinitialize the server library. Do you have any clue what the problem might be?

Thanks in advance

[a comment by an automated account is not shown]
